force-app/main/default/classes/AccountHelper.cls

```diff
@@ -28,19 +28,70 @@ Always ensure that your helper methods are bulk-safe to handle multiple records
 */
 public with sharing class AccountHelper {
     
-    public void setTypeProspect(SObject objects){
-
+    public static void setTypeProspectAfterInsert(List<Account> newAccounts) {
+        List<Account> accountsToUpdate = new List<Account>();
+            for (Account acc : newAccounts) {
+                if (acc.Type == null) {
+                    Account accToUpdateAfterInsert = new Account(Id = acc.Id , Type = 'Prospect');
+                    accountsToUpdate.add(accToUpdateAfterInsert);
```

Comment:
Since this is updating the same record that started the trigger event and not needing to change any related record this should be a before trigger.